huaweicloud/services/acceptance/sfs/data_source_huaweicloud_sfs_turbo_perm_rules_test.go

```diff
@@ -0,0 +1,60 @@
+package sfs
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+)
+
+func TestAccDataSourcePermRules_basic(t *testing.T) {
+	var (
+		dataSource = "data.huaweicloud_sfs_turbo_perm_rules.test"
+		rName      = acceptance.RandomAccResourceName()
+		dc         = acceptance.InitDataSourceCheck(dataSource)
+	)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			acceptance.TestAccPreCheck(t)
+		},
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testDataSourcePermRules_basic(rName),
+				Check: resource.ComposeTestCheckFunc(
+					dc.CheckResourceExists(),
+					dc.CheckResourceExists(),
+					resource.TestCheckResourceAttrSet(dataSource, "rules.#"),
+					resource.TestCheckResourceAttrSet(dataSource, "rules.0.id"),
+					resource.TestCheckResourceAttrSet(dataSource, "rules.0.ip_cidr"),
+					resource.TestCheckResourceAttrSet(dataSource, "rules.0.rw_type"),
+					resource.TestCheckResourceAttrSet(dataSource, "rules.0.user_type"),
+
+					dc.CheckResourceExists(),
+					resource.TestCheckOutput("is_exist_rule", "true"),
```

Comment:
It is unnecessary to repeatedly check whether the rule exists.
Remove the extra `dc.CheckResourceExists()`